service/common/src/test/java/org/apache/polaris/service/persistence/pagination/PageTokenTest.java

```diff
@@ -47,4 +52,49 @@ void testReadEverythingPageToken() {
 
     Assertions.assertThat(PageToken.readEverything()).isEqualTo(PageToken.readEverything());
   }
+
+  @Test
+  void testEntityIdPageToken() {
+    EntityIdPageToken token = new EntityIdPageToken(2);
+
+    Assertions.assertThat(token).isInstanceOf(EntityIdPageToken.class);
+    Assertions.assertThat(token.getId()).isEqualTo(-1L);
+
+    List<String> badData = List.of("some", "data");
+    Assertions.assertThatThrownBy(() -> token.buildNextPage(badData))
```

Comment:
nit: Can we add a comment that states only the following classes are supported by the `buildNextPage` and therefore `updated` functions? I only got this because I read this through - but someone in the future without this context may not get it.